gtk/src/events/mod.rs

```diff
@@ -395,7 +391,9 @@ fn release_upgrade_dialog(state: &mut State, widgets: &EventWidgets) {
     let dialog = UpgradeDialog::new(&state.upgrading_from, &state.upgrading_to);
 
     let answer = dialog.run();
-    dialog.destroy();
+    unsafe {
+        dialog.destroy();
```

Comment:
The gtk-rs release notes mention that the `.close()` method can be used instead of `.destroy()` for closing windows, which avoids `unsafe`.

https://gtk-rs.org/blog/2020/07/04/new-release.html